### ISSUE TITLE
Discussion about CLIs

### ISSUE THREAD

**Issue by joaomcteixeira** [haddocking/haddock3#44]
[Currently](https://github.com/haddocking/haddock3/commit/4e903b25da77ffeef27e3ab74e143bb511bbe6dd) HADDOCK3 has a single command-line interface: `haddock3`, added after #28

However, after #43 I foresee that more CLIs will be needed. Personally, I like the idea of having all functionalities in well structured CLIs instead of separate scripts inside the repository. Yet, the CLI for #42 and #43 is not a user related CLI, it is more of a developer related CLI.

What about we have two main haddoc3 structured CLIs?:

* `haddock3`
* `haddock3-dev`

In the future, CLIs related to structural biology would be placed under `haddock3` and tools related to aid development could be placed under `haddock3-dev`. For example, #43 could be under:

`haddock3-dev cnstree`

In this way the CLI could easily have integrated docs.

It is a bit wordy, but helps converging everything to a single point.

I am still brainstorming this idea, comments are welcomed.

**Comment by rvhonorato** [comment]
No preference from my side; apply rule 1 🙂
